terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl

```diff
@@ -126,26 +126,30 @@ configure_retention() {
 
 configure_automate_infra_views() {
   if chef-automate dev grpcurl automate-gateway list | grep "chef.automate.api.infra_proxy.InfraProxy" &> /dev/null; then
-      chef_server_admin_key=$(cat "/hab/chef-server-admin-key.txt")
+      chef_server_admin_key="$(</hab/chef-server-admin-key.txt tr '\n' ':' | sed 's/:/\\n/g')"
```

Comment:
so we translate the newline to a : and then replace all colons with \\n -- right?

Reply:
what is the < at the beginning for?

Reply:
yes, that's correct

Reply:
@vjeffrey `cat "/hab/chef-server-admin-key.txt | FOO"` and `</hab/chef-server-admin-key.txt FOO` do the same thing. An easier way to understand the second one is to write it the "normal way" which would be `FOO < /hab/chef-server-admin-key.txt`

So this could also be
```
cat /hab/chef-server-admin-key.txt | tr '\n' ':' | sed 's/:/\\n/g'
```
or
```
tr '\n' ':' < /hab/chef-server-admin-key.txt | sed 's/:/\\n/g'
```

This is a bunch of hoop jumping to replace a real newline character with the two characters '\n'.

Reply:
👍 thanks for the explanation!